### PR TITLE
Fix for searching id over array data source

### PR DESCRIPTION
**Pull request by radimSekera**
When condition for filterOne looks like `$condition = ['id' => '25661']; and object structure is ['id' => int, ...] item detail is not working correctly.

Thank you for your time

### CODE REVIEW

**Comment by radimvaculik** [comment]
@radimSekera Please, add some tests.

**Comment by radimvaculik** [comment]
You assume that the key will be called `id`. But what if his name is different? For example `age` and is also `int`?

File: src/DataSource/ArrayDataSource.php [hunk 1]
```diff
@@ -93,6 +93,9 @@ public function filterOne(array $condition): IDataSource
 	{
 		foreach ($this->data as $item) {
 			foreach ($condition as $key => $value) {
+				if ($key === 'id' && is_string($value)) {
+					$value = intval($value);
+				}
 				if ($item[$key] === $value) {
```
Comment (radimvaculik) [inline src/DataSource/ArrayDataSource.php:99]:
```suggestion
				if ($item[$key] == $value) {
```

What about to change to `==` to ignore types?

Comment (radimSekera) [reply src/DataSource/ArrayDataSource.php:99]: Yes i thought of that too, its better idea a should work better. Guess there is no chance of getting like `value = null` and having $item[0] for match. 
Another thing that went over my head is regex check for only numbers in value then convert to to int.

But ignoring types is easiest stuff to do. Thank you

Comment (radimvaculik) [reply src/DataSource/ArrayDataSource.php:99]: Are you going to change the pull request?

Comment (radimSekera) [reply src/DataSource/ArrayDataSource.php]: Pull request updated.

**Comment by radimvaculik** [comment]
@radimSekera Thanks. Great. Could you introduce some more tests, please?

**Comment by radimvaculik** [comment]
@radimSekera ping,

**Pull request closed without merging**